### PR TITLE
Added maxUsableWorkstations to workstation Config

### PR DESCRIPTION
**Pull request by tejal29**
Change-Id: I81e020c0af508ab6376fdd71226de265d65c58ad

fixes: https://github.com/hashicorp/terraform-provider-google/issues/19623

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
workstations: added `maxUsableWorkstations`  field to `WorkstationConfiguration` resource.
```

### CODE REVIEW

File: mmv1/products/workstations/WorkstationConfig.yaml [hunk 1]
```diff
@@ -650,6 +650,11 @@ properties:
     description: |
       Disables support for plain TCP connections in the workstation. By default the service supports TCP connections via a websocket relay. Setting this option to true disables that relay, which prevents the usage of services that require plain tcp connections, such as ssh. When enabled, all communication must occur over https or wss.
     min_version: 'beta'
+  - name: 'maxUsableWorkstations'
+    type: Integer
+    description: |
+      Maximum number of workstations under this configuration a user can have workstations.workstation.use permission on. Only enforced on CreateWorkstation API calls on the user issuing the API request
```
Comment (BBBmau) [inline mmv1/products/workstations/WorkstationConfig.yaml:656]: can you address the lint issues that are coming from this line?:

```hcl
Error: 656:203 [trailing-spaces] trailing spaces
```

**Comment by tejal29** [comment]
@BBBmau Can we merge this in before next release?

File: mmv1/products/workstations/WorkstationConfig.yaml [hunk 2]
```diff
@@ -650,6 +650,12 @@ properties:
     description: |
       Disables support for plain TCP connections in the workstation. By default the service supports TCP connections via a websocket relay. Setting this option to true disables that relay, which prevents the usage of services that require plain tcp connections, such as ssh. When enabled, all communication must occur over https or wss.
     min_version: 'beta'
+  - name: 'maxUsableWorkstations'
+    type: Integer
+    description: |
+      Maximum number of workstations under this configuration a user can have workstations.workstation.use permission on. Only enforced on CreateWorkstation API calls on the user issuing the API request.
+    min_version: 'beta'
```
Comment (SarahFrench) [inline mmv1/products/workstations/WorkstationConfig.yaml:657]: Fly-by review as I just triaged the associated issue - this field appears to be available in the GA version of the API: https://cloud.google.com/workstations/docs/reference/rest/v1/projects.locations.workstationClusters.workstationConfigs#WorkstationConfig.FIELDS.max_usable_workstations

Comment (tejal29) [reply mmv1/products/workstations/WorkstationConfig.yaml:657]: Shd I remove the `min_version` field then?

Comment (tejal29) [reply mmv1/products/workstations/WorkstationConfig.yaml:657]: yes. We are adding support to GA field.

Comment (tejal29) [reply mmv1/products/workstations/WorkstationConfig.yaml:657]: done.

**Review by BBBmau — APPROVED** [review]

**Pull request merged**